### PR TITLE
Improve documentation for `uv init` CLI

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: crates/uv-cli/src/lib.rs [hunk 1]
```diff
@@ -2101,18 +2121,12 @@ pub struct RunArgs {
...
     ///
-    /// Supported formats:
+    /// If the interpreter request is satisfied by a discovered environment, a
+    /// new one will not be created.
     ///
-    /// - `3.10` looks for an installed Python 3.10 using `py --list-paths` on Windows, or
-    ///   `python3.10` on Linux and macOS.
-    /// - `python3.10` or `python.exe` looks for a binary with the given name in `PATH`.
-    /// - `/home/ferris/.local/bin/python3.10` uses the exact Python at the given path.
+    /// See `uv help python` to view supported request formats.
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:2129]: This is coming.

Comment (zanieb) [reply crates/uv-cli/src/lib.rs:2129]: https://github.com/astral-sh/uv/pull/5869

**Comment by zanieb** [comment]
The following are stacked on this pull request

- #5870 
- #5869

File: crates/uv-cli/src/lib.rs [hunk 2]
```diff
@@ -1940,36 +1953,43 @@ impl ExternalCommand {
 #[allow(clippy::struct_excessive_bools)]
 pub struct InitArgs {
     /// The path of the project.
+    ///
+    /// Relative and absolute paths are allowed.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:1957]: I might say "Accepts both relative and absolute paths." Something something passive voice.

File: crates/uv-cli/src/lib.rs [hunk 3]
```diff
@@ -1940,36 +1953,43 @@ impl ExternalCommand {
...
+    ///
+    /// If a `pyproject.toml` is found in any of the parent directories of the
+    /// target path, the project will be added as a workspace member of the
+    /// parent. This behavior can be disabled with `--no-workspace`.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:1961]: "parent, unless `--no-workspace` is provided."?

File: crates/uv-cli/src/lib.rs [hunk 4]
```diff
@@ -1940,36 +1953,43 @@ impl ExternalCommand {
...
-    /// The name of the project, defaults to the name of the directory.
+    /// The name of the project.
+    ///
+    /// Defaults to the name of the directory.
     #[arg(long)]
     pub name: Option<PackageName>,
 
     /// Create a virtual workspace instead of a project.
+    ///
+    /// A virtual workspace does not define dependencies and cannot be
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:1972]: It can include dev dependencies though.

File: crates/uv-cli/src/lib.rs [hunk 5]
```diff
@@ -2101,18 +2121,12 @@ pub struct RunArgs {
...
-    /// The Python interpreter to use to build the run environment.
-    ///
-    /// By default, uv uses the virtual environment in the current working directory or any parent
-    /// directory, falling back to searching for a Python executable in `PATH`. The `--python`
-    /// option allows you to specify a different interpreter.
+    /// The Python interpreter to use for the run environment.
     ///
-    /// Supported formats:
+    /// If the interpreter request is satisfied by a discovered environment, a
+    /// new one will not be created.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:2127]: Could we maybe rephrase as like... "it will be reused" rather than "not created"?

**Review by charliermarsh — APPROVED** [review]
